ci-operator/step-registry/cucushift/installer/rehearse/gcp/cases/valid-dns-private-zone/provision/cucushift-installer-rehearse-gcp-cases-valid-dns-private-zone-provision-commands.sh

```diff
@@ -0,0 +1,186 @@
+
+wrong_zone_name="test-$RANDOM-private-zone"
+expected_err_msg="failed to create install config: platform.gcp.dns.privateZone.name: Invalid value: \"${expected_zone_name}\": found existing private zone ${wrong_zone_name} in project ${PRIVATE_ZONE_PROJECT} with DNS name ${CLUSTER_NAME}.${BASE_DOMAIN}"
+
+cmd="gcloud --project ${PRIVATE_ZONE_PROJECT} dns managed-zones create ${wrong_zone_name} --dns-name ${CLUSTER_NAME}.${BASE_DOMAIN}. --visibility=private --networks ${HOST_PROJECT_NETWORK} --description \"private zone of OCP cluster '${CLUSTER_NAME}'\""
+run_command "${cmd}"
+
+cp "${install_dir}/install-config.yaml" "${install_dir}/install-config.yaml.bak"
+echo "$(date -u --rfc-3339=seconds) - INFO '${INSTALLER_BINARY} create cluster --dir ${install_dir}'"
+${INSTALLER_BINARY} create cluster --dir ${install_dir} &> ${tmp_output} || result=1
+cat "${tmp_output}"
+if ! grep -qF "${expected_err_msg}" "${tmp_output}"; then
+  echo "$(date -u --rfc-3339=seconds) - Scenario A: FAILED, the expected error messages are not found in '.openshift_install.log'."
+else
+  echo "$(date -u --rfc-3339=seconds) - Scenario A: PASSED, found the expected error messages in '.openshift_install.log'."
+  result=0
+fi
+save_artifacts "${install_dir}" "scenario-a"
+
+cmd="gcloud --project ${PRIVATE_ZONE_PROJECT} dns managed-zones delete -q ${wrong_zone_name}"
+run_command "${cmd}"
+
+echo "$(date -u --rfc-3339=seconds) - Scenario B: a DNS private zone of matching name and network, but different dns name is present"
+expected_zone_name="test-$RANDOM-priv-zone"
+create_install_config "${expected_zone_name}" "${PRIVATE_ZONE_PROJECT}" "${install_dir}"
+
+expected_err_msg="failed to create install config: baseDomain: Invalid value: \"${BASE_DOMAIN}\": failed to find matching DNS zone for ${expected_zone_name} with DNS name ${CLUSTER_NAME}.${BASE_DOMAIN}"
+
+cmd="gcloud --project ${PRIVATE_ZONE_PROJECT} dns managed-zones create ${expected_zone_name} --dns-name ${CLUSTER_NAME}.dns.${BASE_DOMAIN}. --visibility=private --networks ${HOST_PROJECT_NETWORK} --description \"private zone of OCP cluster '${CLUSTER_NAME}'\""
+run_command "${cmd}"
+
+cp "${install_dir}/install-config.yaml" "${install_dir}/install-config.yaml.bak"
+echo "$(date -u --rfc-3339=seconds) - INFO '${INSTALLER_BINARY} create cluster --dir ${install_dir}'"
+${INSTALLER_BINARY} create cluster --dir ${install_dir} &> ${tmp_output} || result=1
+cat "${tmp_output}"
+if ! grep -qF "${expected_err_msg}" "${tmp_output}"; then
+  echo "$(date -u --rfc-3339=seconds) - Scenario B: FAILED, the expected error messages are not found in '.openshift_install.log'."
+else
+  echo "$(date -u --rfc-3339=seconds) - Scenario B: PASSED, found the expected error messages in '.openshift_install.log'."
+  result=0
+fi
+save_artifacts "${install_dir}" "scenario-b"
+
+cmd="gcloud --project ${PRIVATE_ZONE_PROJECT} dns managed-zones delete -q ${expected_zone_name}"
+run_command "${cmd}"
```

Comment:
My concern is the zone won't be deleted if any errors occurred in this script after zone was created, as `errexit` was set. Do you consider put it in a trap or deprovision step?

Reply:
Replaced `create cluster` with `create manifests`, so the `delete` commands should work safely. FYI the rehearsal [test](https://qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/view/gs/qe-private-deck/pr-logs/pull/openshift_release/72945/rehearse-72945-periodic-ci-openshift-verification-tests-main-installation-nightly-4.21-gcp-ipi-valid-dns-private-zone-f14/2004491241399521280).

Reply:
I'm not sure, but I think the safe way is put it in the `trap`, we don't know what kind of unexpected error happened in CI system